### PR TITLE
Doc updates to accommodate the Apollo Link merge

### PR DESCRIPTION
**Pull request by hwillson**
This PR accompanies the work completed in https://github.com/apollographql/apollo-client/pull/5412, which brought the Apollo Link core + HTTP functionality directly into the AC core. 

Main changes:

- New basic and advanced HTTP networking sections, that outline how to use Apollo Client without creating a link directly (using `uri`), and how to leverage Apollo Link for more advanced workflows.
- The API docs for [add-on Apollo Links](https://www.apollographql.com/docs/link/) have been adjusted and merged into a new `API - Link` section. 
- Various odd+end updates across the docs to adjust to using the new link sections.

Once this content is merged in and AC 3 is live, the existing https://www.apollographql.com/docs/link/ content should be shutdown/archived.

### CODE REVIEW

File: docs/gatsby-config.js [hunk 1]
```diff
@@ -71,16 +71,32 @@ module.exports = {
             'integrations/webpack',
           ],
           Networking: [
-            'networking/network-layer',
+            'networking/basic-http-networking',
+            'networking/advanced-http-networking',
+            'networking/non-http-networking',
```
Comment (StephenBarlow) [inline docs/gatsby-config.js]: Can cut if we won't be creating a separate article for it at this time

Comment (hwillson) [reply docs/gatsby-config.js]: Great catch - thanks!

**Comment by henryqdineen** [comment]
Is the `uri` option for `ApolloClient` set in stone? We will have our own custom http link and a persisted queries implementation so bundling `HttpLink` and `graphql/language/printer` will be a bundle size regression for us. I have ideas to workaround this issue and it's not a big deal but just thought I would share my thoughts. Thanks and great work on 3.0!

**Comment by hwillson** [comment]
@StephenBarlow I believe we're all set here (I tweaked a few package names and fixed a few broken links since the last approval). I'll go ahead and merge this, but let me know if you want anything changed afterwards. Thanks!

**Pull request merged**